tests/common.go

```diff
@@ -5315,6 +5315,217 @@ func GetSubsetOfSlice[T any](items []T, length int) ([]T, error) {
+				if state == opsapi.StorageRebalanceJobState_PAUSED {
+					return nil, true, nil
+				}
+				return nil, false,
+					fmt.Errorf("Job %v is in paused/pending state", job.GetId())
+			case opsapi.StorageRebalanceJobState_DONE:
+				if state == opsapi.StorageRebalanceJobState_DONE {
+					return nil, true, nil
+				}
+				return nil, false, nil
+			case opsapi.StorageRebalanceJobState_RUNNING:
+				if state == opsapi.StorageRebalanceJobState_RUNNING {
+					return nil, true, nil
+				}
+				return nil, false,
+					fmt.Errorf("job %v has status Running, Summary: %v",
+						job.GetId(),
+						jobResponse.GetSummary().GetWorkSummary())
+			default:
+				if state == jobState {
+					return nil, true,
+						fmt.Errorf("Job not in required state [%v]", state)
+				}
+			}
+		}
+		return nil, false, nil
+	}
+	_, err := task.DoRetryWithTimeout(rebalanceFunc, time.Minute*60, time.Minute*2)
+	if err != nil {
+		return false, err
+	}
+	return true, err
+}
+
+// MakeStoragetoStoragelessNode returns true on converting Storage Node to Storageless Node
+func MakeStoragetoStoragelessNode(n node.Node) error {
+	storageLessNodeBeforePoolDelete := node.GetStorageLessNodes()
+	// Get total list of pools present on the node
+	poolList, err := GetPoolsDetailsOnNode(n)
+	if err != nil {
+		return err
+	}
+
+	lenPools := len(poolList)
+	log.InfoD("total number of Pools present on the Node [%v] is [%d]", n.Name, lenPools)
+
+	// Enter pool maintenance mode before deleting the pools from the cluster
+	err = Inst().V.EnterPoolMaintenance(n)
+	if err != nil {
+		return fmt.Errorf("failed to set pool maintenance mode on node %s : Error [%v]", n.Name, err)
+	}
+
+	time.Sleep(1 * time.Minute)
+	expectedStatus := "In Maintenance"
+	err = WaitForPoolStatusToUpdate(n, expectedStatus)
+	if err != nil {
+		return fmt.Errorf("node [%s] pools are not in status [%s] : Error [%v]", n.Name, expectedStatus, err)
+	}
+
+	// Delete all the pools present on the Node
+	for i := 0; i < lenPools; i++ {
+		err := Inst().V.DeletePool(n, strconv.Itoa(i))
+		if err != nil {
+			return err
+		}
+	}
+
+	err = Inst().V.ExitPoolMaintenance(n)
+	if err != nil {
+		return fmt.Errorf("failed to exit pool maintenance mode on node [%s] Error: [%v]", n.Name, err)
+	}
+
+	err = Inst().V.WaitDriverUpOnNode(n, 5*time.Minute)
+	if err != nil {
+		return fmt.Errorf("volume driver down on node %s with Error: [%v]", n.Name, err)
+	}
+	expectedStatus = "Online"
+	err = WaitForPoolStatusToUpdate(n, expectedStatus)
+	if err != nil {
+		return fmt.Errorf("node %s pools are not in status %s with Error: [%v]", n.Name, expectedStatus, err)
+	}
+
+	storageLessNodeAfterPoolDelete := node.GetStorageLessNodes()
+	if len(storageLessNodeBeforePoolDelete) <= len(storageLessNodeAfterPoolDelete) {
+		return fmt.Errorf("making storage node to storagelessnode failed")
+	}
+	return nil
+}
+
+// IsPksPxOperator returns true if current operator installation is on an EKS cluster
+func IsPksPxOperator() bool {
```

Comment:
can u rename it to IsPksCluster?
 same for other functions also

Reply:
done